lib/openstax_kramdown.rb

```diff
@@ -0,0 +1,64 @@
+require 'addressable/uri'
+
+class Kramdown::Parser::Openstax < Kramdown::Parser::Html
+  ESCAPED_PARENS_MATH_RE = /(.*?)\\\((.*?)\\\)(.*)/
+
+  def secrets
+    @secrets ||= Rails.application.secrets
+  end
+
+  def s3_secrets
+    @s3_secrets ||= secrets.aws[:s3]
+  end
+
+  def region
+    @region ||= s3_secrets[:region]
+  end
+
+  def s3_client
+    @s3_client ||= Aws::S3::Client.new(
+      region: region,
+      credentials: Aws::Credentials.new(s3_secrets[:access_key_id], s3_secrets[:secret_access_key])
+    )
+  end
+
+  def add_text(text, tree = @tree, type = @text_type)
+    matches = ESCAPED_PARENS_MATH_RE.match(text)
+    return super(text, tree, type) if matches.nil?
+
+    super(matches[1], tree, type)
+
+    math = matches[2].strip
+    last = tree.children.last
+    location = (last && last.options[:location] || tree.options[:location])
+    math_el = Kramdown::Element.new(:html_element, 'span', { 'data-math' => math }, location: location)
+    tree.children << math_el
+    super(math, math_el, :text)
+
+    add_text(matches[3], tree, type)
+  end
+
+  def handle_html_start_tag(line, &block)
+    super(line, &block).tap do
+      next if Rails.env.development? || !@options[:attachable]
+      last_el = @tree.children.last
+      next unless last_el.type == :html_element && last_el.value == 'img' && last_el.attr['src']
+
+      uri = URI.parse(last_el.attr['src'])
+      contents = Net::HTTP.get(uri)
```

Comment:
Are these always going to be from openstax hosts? I'm wondering if we should validate the domain...

Reply:
I think that is up to the content team really